cdk_framework/EKS/lib/utils/validate.ts

```
@@ -0,0 +1,107 @@
+const yaml = require('js-yaml')
+
+const supportedFields = new Set(['version', 'cpu_architecture', 'launch_type', 'node_size'])
+const supportedVersions = new Set(['1.18', '1.19', '1.20', '1.21']);
+const supportedCPUArchitectures = new Set(['amd_64', 'arm_64']);
+const supportedLaunchTypes = new Set(['ec2', 'fargate']);
+const supportedNodeSizes = new Set(['medium', 'large', 'xlarge', '2xlarge', '4xkarge', '8xlarge', '12xlarge', '16xlarge', '24xlarge', 'metal']);
+
+
+export function validateClustersConfig(info: Object){
+    const data = Object(info)
+    if(!data['clusters']){
+        throw new Error('No clusters field being filed in the yaml file')
+    }
+    const clusterInfo = data['clusters']
+    for(const [key, value] of Object.entries(clusterInfo)){
+        
+        const val = Object(value)
+        if(Object.keys(val).length !== 4){
+            throw new Error("Didn't set all the fields for the clusters")
+        }
```

Instead of forcing users to set all fields should we just set defaults instead? If there are fields where a default does not make sense we can error out if they are not provided.

---

This was brought up earlier, and I thought it was concluded that there shouldn't be any defaults. I could change this. I feel like we should discuss what the defaults should be, though.